readme.md

```diff
@@ -79,33 +91,28 @@ However, this is quite verbose. Instead, an `@Type` decorator is provided for ne
 
 ```typescript
 class User extends Entity {
-    public name: string = null;
-    public email: string = null;
+    public name!: string;
+    public email!: string;
     @Type(Address)
-    public address: Address = null;
+    public address!: Address;
 }
 ```
 
-If your JSON data comes in with another key, you may specify that manually with:
-```typescript
-@Type(Address, 'json_key')
-```
-
 Note that by default, the `@Type` decorator will assume your JSON comes in snake case. As such,
 ```typescript
 @Type(Address)
-public homeAddress: Address = null;
+public homeAddress!: Address;
 ```
-will assume that the json holds the key `home_address`. If that is not the case, it should be manually specified as the second argument to `@Type`.
+will assume that the json holds the key `home_address`.
 
 #### Note about `Object`
 If your entity has a nested object that is **not** represented by another entity, you can also use `@Type(Object)` to annotate that the object should simply be stored as is.
 
 ### Encoding back to JSON
 
-Entity objects can also be encoded back to a plain JavaScript Object, or as a JSON string. You can call `toJson()` on any entity to convert it to a plain JS object.
+Entity objects can also be encoded back to a plain JavaScript Object. You can call `.toJson()` on any entity to convert it to a plain JS object.
 
-The method defaults to converting your properties to snake case. To prevent this, you can pass `false` as the first argument to `toJson()`. The method also accepts a second boolean argument that lets you specify if the output should instead be as a JSON string. `toJson(true, true)` is identical to `JSON.stringify(toJson(true))`.
```

Comment:
What's the reason for removing this functionality? Do we just no longer want to support it or is there something more to it?

Reply:
Types were so terribly complex with this feature. I could never make the types work in a straightforward way in our apps because of the complex types. See: https://github.com/jobilla/entity/pull/4/files#diff-878e93df041561e9fc7a14735bafefe35fa06a7b1b48db0c1a69d7ff7f4ce882L125-L132

And yeah, as far as I'm aware we have never used this feature in Jobilla apps. And it doesn't make anything considerably easier, too. `JSON.stringify(entity.toJson())` is only slightly more work than `entity.toJson(true, true)`, we might as well get rid of it completely.